tests/conftest.py

```diff
@@ -71,7 +71,7 @@ def jit_compile(source, filepath=None, objname=None):
         Return a specific object (module or subroutine) in :attr:`source`
     """
     if isinstance(source, Sourcefile):
-        filepath = source.filepath if filepath is None else Path(filepath)
+        filepath = source.source if filepath is None else Path(filepath)
```

Comment:
I disagree, `Sourcefile.source` is an object of type `Source` rather than a path. However, the current implementation is also wrong, it should be using `source.path` rather than `source.filepath`: https://github.com/ecmwf-ifs/loki/blob/af8ec9fc47bc529f914969d9e88ed898dc788ebf/loki/sourcefile.py#L60

Reply:
Used path instead of source. #[bc8f27b](https://github.com/ecmwf-ifs/loki/pull/177/commits/bc8f27be4a03c782de64a0c834e3cae530b55d16)